adapters/pubmatic/pubmatic.go

```diff
@@ -637,24 +644,26 @@ func getStringArray(array []interface{}) []string {
 	return aString
 }
 
-// getBidType returns the bid type specified in the response bid.ext
-func getBidType(bidExt *pubmaticBidExt) openrtb_ext.BidType {
+// getMediaType returns the bid type specified in the response bid.ext
+func getMediaTypeForBid(bid *openrtb2.Bid) openrtb_ext.BidType {
 	// setting "banner" as the default bid type
-	bidType := openrtb_ext.BidTypeBanner
-	if bidExt != nil && bidExt.BidType != nil {
-		switch *bidExt.BidType {
-		case 0:
-			bidType = openrtb_ext.BidTypeBanner
-		case 1:
-			bidType = openrtb_ext.BidTypeVideo
-		case 2:
-			bidType = openrtb_ext.BidTypeNative
+	mType := openrtb_ext.BidTypeBanner
+	if bid != nil {
+		switch bid.MType {
+		case openrtb2.MarkupBanner:
+			mType = openrtb_ext.BidTypeBanner
+		case openrtb2.MarkupVideo:
+			mType = openrtb_ext.BidTypeVideo
+		case openrtb2.MarkupAudio:
+			mType = openrtb_ext.BidTypeAudio
+		case openrtb2.MarkupNative:
+			mType = openrtb_ext.BidTypeNative
 		default:
 			// default value is banner
-			bidType = openrtb_ext.BidTypeBanner
+			mType = openrtb_ext.BidTypeBanner
```

Comment:
Do we need default case, as "banner" is the default bid type ?

Reply:
Like other bidders, we can also handle invalid type with an error

```
	default:
		return "", &errortypes.BadServerResponse{
			Message: fmt.Sprintf("Failed to parse media type for bid: \"%s\"", bid.ImpID),
		}
	}
```